widget/entry.go

```diff
@@ -500,7 +504,10 @@ func (e *Entry) TypedKey(key *fyne.KeyEvent) {
 		provider.deleteFromTo(pos, pos+1)
 		e.propertyLock.Unlock()
 	case fyne.KeyReturn, fyne.KeyEnter:
-		if !multiLine {
+		if !multiLine || selectDown {
+			if f := e.OnSubmitted; f != nil {
+				f(text)
+			}
 			return
```

Comment:
I don't think we want to return if `e.OnSubmitted == nil`, because creating a new line with shift + enter will not work in a situation where the developer hasn't set any e.OnSubmitted function.

Reply:
Good point, is inserting a newline with shift held a use case we want/need to support?

Reply:
It worked like that before and feels a bit strange to not have in my opinion.

Reply:
Or maybe not? It would be a discrepancy between how they work. Should we perhaps land it as is and then tweak it in the future when more people have a chance of chiming in?

Reply:
Should we not maintain old behaviour unless we intend to change it?

Reply:
Done